### ISSUE TITLE
A Bug in Restart

### ISSUE THREAD

**Issue by Yin-YinjianZhao** [ECP-WarpX/WarpX#1351]
I have tested the restart of WarpX and found a possible bug.

For a simple run (based on a setup that I used recently with Remi for collisions), the restart looks good.
Both plotfiles and reduced diags seem to work fine.
This run uses a moving window in the lab frame, and no field is solved.
The input is attached.
[input1.txt](https://github.com/ECP-WarpX/WarpX/files/5246650/input1.txt)
Directly run using the input will give a result without restart until time step = 20.
To use restart, change `max_step=10` first and run, then change `max_step=20` and uncomment `amr.restart = ./diags/check00010` to load the restart file.
Note that I disabled the collision module, because it uses random numbers, which won't give us exact results after a restart.
![e](https://user-images.githubusercontent.com/56095356/93619459-69442500-f98d-11ea-87bd-d626ecf48973.png)
The above figure shows the matched results.

However, when I change `max_step = 200`, `amr.restart = ./diags/check00100`, and
```
103 diagnostics.diags_names = diag check
104 diag.period = 10
105 diag.diag_type = Full
106 check.period = 100
107 check.diag_type = Full
108 check.format = checkpoint
```
I got zero beam particles in plotfiles after the restart.
The reduced diag looks good.
[input2.txt](https://github.com/ECP-WarpX/WarpX/files/5246661/input2.txt)
![e2](https://user-images.githubusercontent.com/56095356/93619646-b0cab100-f98d-11ea-9ed3-aeeb393b390d.png)
Also noticed the `domain_dimensions` changes to 8 8 2 after the restart, it should be 8 8 8.
```
./restart_yes/diags/diag00100
yt : [INFO     ] 2020-09-04 15:41:57,513 Parameters: current_time              = 9.02734313224906e-13
yt : [INFO     ] 2020-09-04 15:41:57,513 Parameters: domain_dimensions         = [8 8 8]
yt : [INFO     ] 2020-09-04 15:41:57,513 Parameters: domain_left_edge          = [-0.00015   -0.00015    0.0001125]
yt : [INFO     ] 2020-09-04 15:41:57,513 Parameters: domain_right_edge         = [0.00015   0.00015   0.0004125]
number of x:  100000
./restart_yes/diags/diag00110
yt : [INFO     ] 2020-09-04 15:41:57,912 Parameters: current_time              = 9.930077445473966e-13
yt : [INFO     ] 2020-09-04 15:41:57,912 Parameters: domain_dimensions         = [8 8 2]
yt : [INFO     ] 2020-09-04 15:41:57,912 Parameters: domain_left_edge          = [-0.00015   -0.00015    0.0001125]
yt : [INFO     ] 2020-09-04 15:41:57,912 Parameters: domain_right_edge         = [0.00015   0.00015   0.0001875]
number of x:  0
```

**Comment by Yin-YinjianZhao** [comment]
I did another test using `Example/Physics_applications/laser_acceleration/inputs_3d`, and found a similar error.
[inputs_3d.txt](https://github.com/ECP-WarpX/WarpX/files/5247074/inputs_3d.txt)
After the restart, `domain_dimensions` changed to `64 64 22`, but it should be `64 64 512`.
```
yes$ python plot.py
./diags/diag00500
yt : [INFO     ] 2020-09-18 10:11:02,590 Parameters: current_time              = 2.17191380649e-13
yt : [INFO     ] 2020-09-18 10:11:02,590 Parameters: domain_dimensions         = [ 64  64 512]
yt : [INFO     ] 2020-09-18 10:11:02,590 Parameters: domain_left_edge          = [-3.000000e-05 -3.000000e-05  9.078125e-06]
yt : [INFO     ] 2020-09-18 10:11:02,590 Parameters: domain_right_edge         = [3.0000000e-05 3.0000000e-05 7.7078125e-05]
./diags/diag00600
yt : [INFO     ] 2020-09-18 10:11:03,198 Parameters: current_time              = 2.60629656779e-13
yt : [INFO     ] 2020-09-18 10:11:03,198 Parameters: domain_dimensions         = [64 64 22]
yt : [INFO     ] 2020-09-18 10:11:03,198 Parameters: domain_left_edge          = [-3.000000e-05 -3.000000e-05  2.209375e-05]
yt : [INFO     ] 2020-09-18 10:11:03,198 Parameters: domain_right_edge         = [3.0000000e-05 3.0000000e-05 2.5015625e-05]
```
Also, the electrons are gone after the restart.
For reduced diags, this time, we can see a difference between no restart and restart.
![e](https://user-images.githubusercontent.com/56095356/93626824-de692780-f998-11ea-8af8-6dcf012bd4fb.png)
Note I'm evaluating the emittance for plasma electrons here, since it is the only species in this example.

**Comment by Yin-YinjianZhao** [comment]
@atmyers @WeiqunZhang 
Hi Andrew and Weiqun, based on the testing results shown above, do you have an idea where in the code the bug would be? From the 1st test, we may feel the bug is in the diagnostic of plotfiles, since reduced diagnostic works fine. But for the second test, the reduced diagnostic also produces a difference, so I'm not sure anymore.

**Comment by Yin-YinjianZhao** [comment]
I did another test using `Example/Physics_applications/laser_acceleration/inputs_3d`, and a similar error.

**Comment by Yin-YinjianZhao** [comment]
[plot.txt](https://github.com/ECP-WarpX/WarpX/files/5247285/plot.txt)
Attached a yt script to get the data, however I use gnuplot to plot, but still just in case this script can help.

**Comment by Yin-YinjianZhao** [comment]
Fixed by @RevathiJambunathan 's PR #1354 .
Thank you!
Now, all the tests mentioned above produce good matched results!
